doc/changelog/04-tactics/13403-occs_nums_nat.rst

```diff
@@ -0,0 +1,8 @@
+- **Removed:**
+  "at @occ_nums" clauses in tactics such as autounfold
+  no longer allow negative values.  A "-" before the
+  list (for set complement) is still supported.  Ex: "at 1 -2"
+  is no longer supported but "at -1 2" is.  Does not
+  affect rewrite * or setoid_rewrite.
```

Comment:
As far as I understand `rewrite` and `setoid_rewrite` have *exactly* the same issue, so I suggest restricting the syntax accepted by `rewrite` and `setoid_rewrite` in the same way. But even if you do not implement that, this changelog should not mention these tactics anyway because we should certainly not advertise this problematic syntax that we should remove eventually.

Reply:
It's not exactly the same.  Those two use different nonterminals to specify the occurrences and slightly different code.  #13391 recommends changing them so they use the same nonterminal  and have the same functionality as `unfold` et al.  Tweaking `occurrences` and function `occurrences_of` in the mlgs is a band aid when something more is needed.  (Note that I rename mlg `occurrences` to `rewrite_occs` in the doc and similarly `clause_dft_concl` in the mlg to `occurrences` in the doc.)

I removed the sentence.

Reply:
FWIW, I found the following tactics have `int_or_var` where it appears not to be useful (some give error messages, others don't):
constructor, econstructor, do, timeout, hresolve_core, hget_var, auto, info_auto, eauto, info_eauto, dfs eauto, probably bfs eauto, probably typeclasses eauto, psatz_Z, psatz_R, psatz_Q, psatz

debug_auto and debug_eauto give slightly different results for negative numbers, but negatives don't make sense.

I'll create a separate PR for those.